src/Integration.Vsix.UnitTests/Analysis/AnalysisConfigMonitorTests.cs

```diff
@@ -42,6 +44,7 @@ public void WhenUserSettingsChange_StandaloneMode_AnalysisIsRequested()
 
             // Should re-analyse
             builder.AssertAnalysisIsRequested();
+            builder.AssertSwitchedToBackgroundThread();
```

Comment:
I didn't think adding another test for just checking on background task made sense. So I've added the checks in scenarios where it should run on background.